app/src/main/java/org/luxoft/sdl_core/BluetoothLongWriter.java

```diff
@@ -2,16 +2,20 @@
 
 import android.util.Log;
 
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
 import java.nio.ByteBuffer;
 import java.util.ArrayList;
 import java.util.Arrays;
+import java.util.zip.DataFormatException;
+import java.util.zip.Deflater;
 
 public class BluetoothLongWriter {
     private static final String TAG = BluetoothLongWriter.class.getSimpleName();
 
     private int mMtu = 23;
-    private final int mFramesCountBytes = 4;
-    private final int mReserved = 3 + mFramesCountBytes; // 3 bytes for internal information and 4 bytes for frames length
+    private final int mFramesCountBytes = 5; // 4 bytes for frames length and 1 byte for compression flag
+    private final int mReserved = 3 + mFramesCountBytes; // 3 bytes for internal information
     private boolean mSendInProgress = false;
     private ArrayList<byte[]> mMessagesToSend = new ArrayList<>();
```

Comment:
Maybe we can use ConcurrentLinkedQueue instead and use method poll() for line 54 - 55?

Reply:
thanks. done in 90e723f119487ab7640493894415e6e1d860a14e